### ISSUE TITLE
DiffEqFlux 0.6.0, 0.5.0 and 0.5.1 do not compile on julia 1.0

### ISSUE THREAD

**Issue by DaanVanHauwermeiren** [JuliaDiffEq/DiffEqFlux.jl#81]
I have found that under `Julia 1.0` `DiffEqFlux` does not compile for versions `0.6.0`, `0.5.1` and `0.5.0`.
The combinations `Julia 1.2.0` + `DiffEqFlux 0.6.0` and `Julia 1.0 `+ `DiffEqFlux 0.4.0` work just fine.

Tested on `Ubuntu 14.04.6 LTS` and `Ubuntu 18.04.2 LTS`. Both systems give the same results.

Tested on a clean environment with only `DiffEqFlux` as installed package.

Error dump:

```
Precompiling DiffEqFlux
[ Info: Precompiling DiffEqFlux [aae7a2af-3d4f-5e19-a356-7da93b79d9d0]
ERROR: LoadError: LoadError: syntax: invalid function name "Morris <: GSAMethod"
Stacktrace:
 [1] include at ./boot.jl:317 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1038
 [3] include at ./sysimg.jl:29 [inlined]
 [4] include(::String) at /home/jos/.julia/packages/DiffEqSensitivity/xlGSs/src/DiffEqSensitivity.jl:3
 [5] top-level scope at none:0
 [6] include at ./boot.jl:317 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1038
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:319 [inlined]
 [11] eval(::Expr) at ./client.jl:389
 [12] top-level scope at ./none:3
in expression starting at /home/jos/.julia/packages/DiffEqSensitivity/xlGSs/src/morris_sensitivity.jl:1
in expression starting at /home/jos/.julia/packages/DiffEqSensitivity/xlGSs/src/DiffEqSensitivity.jl:17
ERROR: LoadError: Failed to precompile DiffEqSensitivity [41bf760c-e81c-5289-8e54-58b1f1f8abe2] to /home/jos/.julia/compiled/v1.0/DiffEqSensitivity/02xYn.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] top-level scope at none:2
 [12] eval at ./boot.jl:319 [inlined]
 [13] eval(::Expr) at ./client.jl:389
 [14] top-level scope at ./none:3
in expression starting at /home/jos/.julia/packages/DiffEqFlux/Cufen/src/DiffEqFlux.jl:3
ERROR: Failed to precompile DiffEqFlux [aae7a2af-3d4f-5e19-a356-7da93b79d9d0] to /home/jos/.julia/compiled/v1.0/DiffEqFlux/BdO4p.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] precompile(::Pkg.Types.Context) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:489
 [5] do_precompile!(::Dict{Symbol,Any}, ::Array{String,1}, ::Dict{Symbol,Any}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/REPLMode.jl:586
 [6] #invokelatest#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:686
 [7] invokelatest(::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:685
 [8] do_cmd!(::Pkg.REPLMode.PkgCommand, ::REPL.LineEditREPL) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/REPLMode.jl:542
 [9] #do_cmd#30(::Bool, ::Function, ::REPL.LineEditREPL, ::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/REPLMode.jl:507
 [10] do_cmd at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/REPLMode.jl:503 [inlined]
 [11] (::getfield(Pkg.REPLMode, Symbol("##41#44")){REPL.LineEditREPL,REPL.LineEdit.Prompt})(::REPL.LineEdit.MIState, ::Base.GenericIOBuffer{Array{UInt8,1}}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/REPLMode.jl:842
 [12] #invokelatest#1 at ./essentials.jl:686 [inlined]
 [13] invokelatest at ./essentials.jl:685 [inlined]
 [14] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/LineEdit.jl:2261
 [15] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:1029
 [16] run_repl(::REPL.AbstractREPL, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:191
 [17] (::getfield(Base, Symbol("##720#722")){Bool,Bool,Bool,Bool})(::Module) at ./logging.jl:311
 [18] #invokelatest#1 at ./essentials.jl:686 [inlined]
 [19] invokelatest at ./essentials.jl:685 [inlined]
 [20] macro expansion at ./logging.jl:308 [inlined]
 [21] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:330
 [22] exec_options(::Base.JLOptions) at ./client.jl:242
 [23] _start() at ./client.jl:421

```

**Comment by ChrisRackauckas** [comment]
Yeah, there's not much we can do about this one. 1.0 has fundamental problems here. We need a newer Julia for some of the code generation stuff to work properly, and the compile times will be horrendous on 1.0. Besides, 1.3 or 1.4 will be the new LTS release soon, so I'm not going to put too much time into fixing anything with 1.0 because it's just time.

**Comment by DaanVanHauwermeiren** [comment]
Thanks for the info @ChrisRackauckas! I'll keep an eye open for the new Julia LTS release.

**Comment by ChrisRackauckas** [comment]
Handled the original issue: https://github.com/JuliaDiffEq/DiffEqSensitivity.jl/pull/116 , but it looks like 1.0 still has other compiler issues.

**Comment by ChrisRackauckas** [comment]
The newest release is now using Zygote and tons of things like that, so it definitely won't get v1.0 compatible. I think we need to just suggest not using the current LTS and we are just going to keep moving forward. Sorry if this causes any issues, but the fix is to just use Julia v1.3.
